datacube/scripts/ingest.py

```diff
@@ -304,6 +304,8 @@ def _index_datasets(index, results):
             extra_args['storage_metadata'] = datasets.attrs['storage_metadata']
 
         for dataset in datasets.values:
+            if 'storage_metadata' in extra_args:
```

Comment:
Let's call it  `driver_data` see my next comment for justification. The whole dataset document is a kind of "storage metadata", so let's be more specific here.

Reply:
Done